toolchain/lowering/lowering_handle.cpp

```diff
@@ -79,14 +79,38 @@ auto LoweringHandleArrayValue(LoweringFunctionContext& context,
 auto LoweringHandleAssign(LoweringFunctionContext& context,
                           SemIR::NodeId /*node_id*/, SemIR::Node node) -> void {
   auto [storage_id, value_id] = node.GetAsAssign();
+  auto storage_type_id = context.semantics_ir().GetNode(storage_id).type_id();
+
   // We can assign from either a value expression or a by-copy initializing
   // expression. In either case, the value of the source is the value
   // representation of the target.
   // TODO: Should we use different semantic nodes for those operations?
-  // TODO: Handle the case where the value representation is not the object
-  // representation.
-  context.builder().CreateStore(context.GetLocal(value_id),
-                                context.GetLocal(storage_id));
+  switch (auto rep = SemIR::GetValueRepresentation(context.semantics_ir(),
+                                                   storage_type_id);
+          rep.kind) {
+    case SemIR::ValueRepresentation::None:
+      break;
+    case SemIR::ValueRepresentation::Copy:
+      context.builder().CreateStore(context.GetLocal(value_id),
+                                    context.GetLocal(storage_id));
+      break;
+    case SemIR::ValueRepresentation::Pointer: {
+      auto& layout = context.llvm_module().getDataLayout();
+      auto* type = context.GetType(storage_type_id);
+      auto size = layout.getTypeAllocSize(type);
```

Comment:
You might consider inlining this since it's only used once.

Reply:
Done.